cli_tools/common/utils/logging/service/logger.go

```diff
@@ -229,7 +232,10 @@ func (l *Logger) runWithServerLogging(function func() (Loggable, error),
 		l.logStart()
 	}()
 
-	loggable, err := function()
+	// Execute the closure and ensure that panic won't crash the program.
+	r := &panicSafeRunner{inner: function}
+	r.runInner()
+	loggable, err := r.loggable, r.err
```

Comment:
why not just return loggable & err from runInner()?

Reply:
I didn't think it would be possible, but I'm glad you suggested it! Updated to remove the intermediate struct.